internal/settings/providers/netcup/provider.go

```diff
@@ -0,0 +1,145 @@
+package netcup
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+
+	"github.com/qdm12/ddns-updater/internal/models"
+	"github.com/qdm12/ddns-updater/internal/settings/constants"
+	"github.com/qdm12/ddns-updater/internal/settings/errors"
+	"github.com/qdm12/ddns-updater/internal/settings/utils"
+	"github.com/qdm12/ddns-updater/pkg/publicip/ipversion"
+)
+
+type Provider struct {
+	customerNumber string
+	domain         string
+	host           string
+	ipVersion      ipversion.IPVersion
+	apiKey         string
+	password       string
+}
+
+func New(data json.RawMessage, domain, host string,
+	ipVersion ipversion.IPVersion) (p *Provider, err error) {
+	extraSettings := struct {
+		CustomerNumber string `json:"customer_number"`
+		APIKey         string `json:"api_key"`
+		Password       string `json:"password"`
+	}{}
+	if err := json.Unmarshal(data, &extraSettings); err != nil {
+		return nil, err
+	}
+	p = &Provider{
+		domain:         domain,
+		host:           host,
+		ipVersion:      ipVersion,
+		customerNumber: extraSettings.CustomerNumber,
+		apiKey:         extraSettings.APIKey,
+		password:       extraSettings.Password,
+	}
+	if err := p.isValid(); err != nil {
+		return nil, err
+	}
+	return p, nil
+}
+
+func (p *Provider) isValid() error {
+	switch {
+	case p.customerNumber == "":
+		return errors.ErrEmptyCustomerNumber
+	case p.apiKey == "":
+		return errors.ErrEmptyAppKey
+	case p.password == "":
+		return errors.ErrEmptyPassword
+	case p.host == "*":
+		return errors.ErrHostWildcard
```

Comment:
Precise in the documentation it cannot be a wildcard.
Also precise it can be `@`